QtCollider/QcApplication.cpp

```diff
@@ -104,6 +104,29 @@ QcApplication::~QcApplication() {
     _mutex.unlock();
 }
 
+void QcApplication::createMenu() {
+    _mainMenu = QSharedPointer<QMenuBar>(new QMenuBar(0));
+
+#ifdef Q_OS_MAC
+    // macOS registers cmd+q on menu bars by default. Here we register a handler for cmd+q that does nothing
+    // and we disable the Quit menu by default
+    QMenu* menu;
+    QAction* action;
+    action = new QAction(tr("Catch CMD+Q on MacOS X"));
+    action->setMenuRole(QAction::QuitRole);
+    QObject::connect(action, SIGNAL(triggered()), this, SLOT(onQuit()));
+    menu = new QMenu(tr("&File"));
+    menu->addAction(action);
+    _mainMenu->addMenu(menu);
+    action->setEnabled(false);
```

Comment:
this block would be a lot clearer as:

```cpp
    auto* action = new QAction(...);
    action->setMenuRole
    action->setEnabled
    QObject::connect
    auto* menu = new QMenu
    menu->addAction
    _mainMenu->addMenu
```

in general - declare variables at the line they're needed and no earlier, use auto when the right-hand side type is clear, and group related calls together

Reply:
Again, thanks, this is all very good to know !